novice-goals.Rmd

```diff
@@ -0,0 +1,210 @@
+# Outline {#outline}
+
+This outline describes the questions that the novice courses on R and Python will answer.
+The advanced course can then assume that learners have hands-on experience with these topics
+but nothing more.
+
+## Data manipulation
+
+-   How can I read tabular data into a program?
+    -   CSV
+    -   reading files
+-   How can I select subsets of my data?
+    -   select
+    -   filter
+    -   Boolean conditions
+-   How can I calculate new values?
+    -   mutate
+    -   ifelse
+-   How can I tell what's gone wrong in my programs?
+    -   reading error messages
+    -   the difference between syntax and runtime errors
+-   How can I operate on subsets of my data?
+    -   group
+    -   summarize
+-   How can I work with two or more datasets?
+    -   join
+-   What *isn't* included?
+    -   anything other than reasonably tidy tabular data
+    -   map
+    -   loops and conditionals
+
+## Programming
+
+-   What are the different ways I can interact with software?
+    -   console
+    -   scripts
+    -   libraries/packages?
+-   How can I find and view help?
+    -   In the IDE
+    -   Stack Overflow
+-   How can I ask for help?
+    -   creating a reproducible example (reprex)
+-   How do I install software?
+    -   package manager
+-   How can I inspect data while I'm working on it?
+    -   table viewers
+-   What *isn't* included?
+    -   code browsers, multiple cursors, and other fancy IDE tricks
+    -   virtual environments
+    -   debuggers
+
+## Plotting
+
+-   What are the core elements of every plot?
+    -   data
+    -   geometry
+    -   mapping
+-   What kinds of plot should I use?
+    -   Abela's Rules
+-   How can I create different kinds of plots?
```

Comment:
Can we include some brief theory here on plot groupings (e.g. categorical vs quantitative, quantitative vs quantitative, and distributions/counts).

Reply:
@cwickham what do you think?

Reply:
I'm a fan of getting novices familiar with a few common plot forms (e.g. those built on points, lines and bars) and a small number of general guidelines.  My current thinking is the most useful advice is "important comparisons are easiest when they are on the same scale and close together".

I think "Abela's Rules" looks a bit too complex for a novice (and I disagree strongly with some suggestions, e.g. a 3D area chart should never be used).  I do however like the idea of focusing on purpose - "What would you like to show?".

I also like the approach in [Fundamentals of Data Visualization](https://serialmentor.com/dataviz/directory-of-visualizations.html) where the focus is on what you are visualizing (amounts, proportions, trends, distributions, associations etc.), but again this probably goes into way more detail and options than a novice needs.

The breakdown in R4DS's [Exploratory Data Analysis](https://r4ds.had.co.nz/exploratory-data-analysis.html) of questions into those about variation and co-variation also works quite well.

So, I guess after introducing some common forms, I'd like to see us give some guidance of choosing plots to match questions of interest - with the emphasis being first on "what are you trying to show/see?", and then broken down into a few strategies that depend on the types of variables involved.

Will we need to teach these learners what to look for as well?  I.e. can we assume they know how to read a scatterplot?

Reply:
Taken out Abela's Rules, added "which to choose" under plotting.